Plugins/Public/base_plugin/Main.cpp

```diff
@@ -2021,92 +2021,79 @@ void BaseDestroyed(uint space_obj, uint client)
 	}
 }
 
-void __stdcall HkCb_AddDmgEntry(DamageList *dmg, unsigned short p1, float& damage, enum DamageEntry::SubObjFate fate)
+void __stdcall HkCb_AddDmgEntry(DamageList *dmg, unsigned short sID, float& newHealth, enum DamageEntry::SubObjFate fate)
 {
 	returncode = DEFAULT_RETURNCODE;
-	if (iDmgToSpaceID && dmg->get_inflictor_id())
+	if (!iDmgToSpaceID || !dmg->is_inflictor_a_player())
```

Comment:
`|| !dmg->get_inflictor_id()`? or are we sure it's redundant given the `is_inflictor_a_player` check?

Reply:
also, need to do `returncode = SKIPPLUGINS_NOFUNCTIONCALL` and possibly `iDmgToSpaceID = 0` if `!dmg->is_inflictor_a_player`

Reply:
Good catch, the method switch was an oversight, it would cause NPCs to deal regular damage to POBs.
No need for `returncode = SKIPPLUGINS_NOFUNCTIONCALL`, that would stop ALL damage events from being processed, we only want to do that once we confirm we're dealing with a POB, and that it should not get affected whatsoever.

Added the iDmgToSpaceID = 0 in an appropriate place.